src/routing_table/mod.rs

```diff
@@ -694,4 +701,132 @@ mod tests {
         assert!(table.is_empty());
         assert_eq!(table.iter().count(), 0);
     }
+
+    #[test]
+    fn test_routing_groups() {
+        // Use replicable random numbers to initialse a table:
+        use rand::{Rng, SeedableRng, XorShiftRng};
+        let mut rng: XorShiftRng = SeedableRng::from_seed([1315, 30, 61894, 315]);
+        let our_name = rng.next_u32();
+        let mut table = RoutingTable::new(our_name, 8);
+        let mut unknown_distant_name = None;
+
+        for _ in 0..1000 {
+            let new_name = rng.next_u32();
+            // Try to add new_name. We double-check the output to test this too.
+            match table.add(new_name) {
+                Err(Error::AlreadyExists) => {
+                    assert!(table.iter().any(|u| *u == new_name));
+                    // skip
+                }
+                Err(Error::PeerNameUnsuitable) => {
+                    assert!(table.groups.keys().all(|p| !p.matches(&new_name)));
+                    // We should get a few of these. Save one for tests, but otherwise ignore.
+                    unknown_distant_name = Some(new_name);
+                }
+                Err(_) => {
+                    assert!(false); // no other errors should be possible
```

Comment:
`assert!(false)` will give the line number which is enough... but yes, `panic` is probably the correct way to do this.